File: examples/swift/Sources/JsonParserLib/JsonParser.swift
import Foundation

/// Minimal JSON configuration parser used to demonstrate consuming the
/// json2vars-setter matrix file in a Swift project. It uses only Foundation
/// (`JSONSerialization`), so no package dependencies are required.
public enum JsonParser {
    /// Parse `text` (JSON) into a dictionary. Returns `nil` on parse failure.
    public static func parseConfig(_ text: String) -> [String: Any]? {
        guard let data = text.data(using: .utf8) else { return nil }
        let object = try? JSONSerialization.jsonObject(with: data)
        return object as? [String: Any]
    }
}

File: examples/flutter/lib/json_parser.dart
// Minimal JSON configuration parser used to demonstrate consuming the
// json2vars-setter matrix file in a Flutter package. It uses only the Dart SDK
// (dart:convert), so no extra package dependencies are required.

import 'dart:convert';

/// Parse [contents] (JSON text) into a map. Returns `null` on parse failure.
Map<String, dynamic>? parseConfig(String contents) {
  try {
    return jsonDecode(contents) as Map<String, dynamic>;
  } catch (_) {
    return null;
  }
}
